src/edit/PosterFrameUpload.vue

```diff
@@ -0,0 +1,157 @@
+          :is-selected="!image && (selectedIndex === index)" />
+      </div>
+      <p class="mt-3 mb-0 text-left">
+        or upload an image from your computer
+      </p>
+      <div class="mt-3 text-left">
+        <upload-btn
+          @change="upload"
+          label="Upload image"
+          accept="image/jpeg"
+          small depressed>
+          <template #icon>
+            <v-icon>mdi-upload</v-icon>
+          </template>
+        </upload-btn>
+        <p :class="{ 'error--text': isError }" class="mt-1 text-caption">
+          Poster frame must be under 500KB.
+        </p>
+      </div>
+    </template>
+    <template #actions>
+      <v-btn @click="reset" color="primary" text>Close</v-btn>
+      <v-btn @click="save" color="success" text>Save</v-btn>
+    </template>
+  </tailor-dialog>
+</template>
+
+<script>
+import isNil from 'lodash/isNil';
+import PosterFrame from './PosterFrame.vue';
+import TailorDialog from '../tce-core/TailorDialog.vue';
+import take from 'lodash/take';
+import UploadBtn from './UploadBtn.vue';
+
+const MAX_SIZE = 500000; // 500 KB
+const CUSTOM_POSTER_FRAME_INDEX = 4;
+
+export default {
+  name: 'poster-frame-upload',
+  props: {
+    id: { type: String, default: null },
+    playable: { type: Boolean, default: false },
+    selectedPosterFrameIndex: { type: Number, default: 0 },
+    posterFrames: { type: Array, default: () => ([]) }
+  },
+  data: () => ({
+    dialog: false,
+    selectedIndex: null,
+    image: null,
+    isError: false
+  }),
+  computed: {
+    isDisabled: ({ id: videoId, playable }) => !videoId || !playable,
+    generatedPosterFrames: ({ posterFrames }) => take(posterFrames, 4),
+    customPosterFrame: ({ image, posterFrames }) => {
+      return image || posterFrames[CUSTOM_POSTER_FRAME_INDEX];
+    },
+    currentPosterFrame() {
+      const { customPosterFrame, posterFrames, selectedPosterFrameIndex } = this;
+      return customPosterFrame || posterFrames[selectedPosterFrameIndex];
+    }
+  },
+  methods: {
+    reset() {
+      this.dialog = false;
+      this.image = null;
+      this.selectedIndex = this.selectedPosterFrameIndex;
+      this.isError = false;
+    },
+    upload(e) {
+      this.isError = false;
+      this.image = null;
+      const [file] = e.target.files;
+      if (file.size > MAX_SIZE) {
+        this.isError = true;
+        return;
+      }
+      const fileReader = new FileReader();
+      fileReader.readAsDataURL(file);
+      fileReader.addEventListener('load', e => {
+        this.image = e.target.result;
+      });
+    },
+    save() {
+      const { image, selectedIndex, selectedPosterFrameIndex } = this;
+      if (image) {
+        this.$emit('save', {
+          video: {
+            customPosterFrame: image
+          }
+        });
+      } else {
+        this.$emit('save', {
+          video: {
+            posterFrameNumber: isNil(selectedIndex) ? selectedPosterFrameIndex : selectedIndex
+          }
+        });
+      }
```

Comment:
If we derive state from props as mentioned above, I think this could be rewritten to something like:
```suggestion
        this.$emit('save', {
          video: {
            ...image ? { customPosterFrame: image } : { posterFrameNumber: selectedIndex }
          }
        });
      }
```